src/dataflow/analysis/graph-internals.ts

```diff
@@ -72,16 +72,22 @@ export class Flow {
 
   /** Evaluates the given CheckCondition against the current Flow. */
   private checkCondition(condition: FlowCondition): boolean {
+    let result: boolean;
     switch (condition.type) {
       case 'node':
-        return this.nodeIds.has(condition.value);
+        result = this.nodeIds.has(condition.value);
+        break;
       case 'edge':
-        return this.edgeIds.has(condition.value);
+        result = this.edgeIds.has(condition.value);
+        break;
       case 'tag':
-        return this.tags.has(condition.value);
+        result = this.tags.has(condition.value);
+        break;
       default:
         throw new Error('Unknown condition type.');
     }
+    // Flip the result if the check condition was negated.
+    return condition.negated ? !result : result;
```

Comment:
I think this is equivalent to `condition.negated === !result`.
Maybe `condition.expectation === result` would be a neater check, though I think this is just as clear.

Reply:
I think I'll leave it as the ternary operator, that seems the easiest to understand to me.